modules/host/contractmanager/storagefoldershrink_bench_test.go

```diff
@@ -0,0 +1,38 @@
+package contractmanager
+
+import (
+	"testing"
+)
+
+// BenchmarkUnsetManySectors checks that unsetting a bunch of sectors
+// individually takes an acceptable amount of time. Millions should be possible
+// in well under a second.
+//
+// My laptop is doing 10e6 in 1.2 seconds. This is on the edge of too slow, but
+// overall it's close enough.
+func BenchmarkUnsetManySectors(b *testing.B) {
+	// Create a uint64 that has all of the bits set.
+	base := uint64(0)
+	base--
+
+	// Benchmark how long it takes to unset the bits.
+	for i := 0; i < b.N; i++ {
+		// Create a usage array with all bits set.
+		b.StopTimer()
+		usageArray := make([]uint64, 10e6)
+		for i := 0; i < len(usageArray); i++ {
+			usageArray[i] = base
+		}
+		b.StartTimer()
```

Comment:
It might be optimized out though if a known zero value is being set to zero again. I'd rather keep it this way.